Fail to build with cmake 4

Hello.

On Arch Linux the binary either fail to [build](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=stepmania&id=0a9c58941523cfe6d1da58b083ea4a892d70d1ac#n48) with the following error:

```
$ cmake \
  -DCMAKE_INSTALL_PREFIX=/opt \
  -DWITH_FULL_RELEASE=YES \
  -DWITH_PORTABLE_TOMCRYPT=NO \
  -DWITH_SYSTEM_FFMPEG=YES \
  -DWITH_SYSTEM_MAD=YES \
  -DWITH_SYSTEM_OGG=YES \
  -DWITH_SYSTEM_JPEG=YES \
  -DWITH_SYSTEM_PNG=YES \
  -DWITH_SYSTEM_GLEW=YES \
  -DWITH_SYSTEM_TOMMATH=YES \
  -DWITH_SYSTEM_TOMCRYPT=YES \
  -DWITH_SYSTEM_JSONCPP=YES \
  -DWITH_SYSTEM_PCRE=YES \
  -DWITH_SYSTEM_ZLIB=YES \
  -Wno-dev \
  ..

CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.

```

Or, if the `-DCMAKE_POLICY_VERSION_MINIMUM=3.5` parameter is added, the game doesn't start with the following error:

```
$ stepmania
StepMania5.1.0
Compiled 20250727 @ 14:07:06 +0200 (build 0a9c589)
Log starting 2025-07-27 14:09:51
Couldn't load driver gtk: dlopen(): /opt/stepmania-5.1/GtkModule.so: undefined symbol: _Z13CreateSurfaceiiijjjj
Error: Couldn't open any loading windows.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5)
 
 project(StepMania)
 

--- a/extern/CMakeProject-png.cmake
+++ b/extern/CMakeProject-png.cmake
@@ -20,6 +20,13 @@ else()
       "libpng/pngwtran.c"
       "libpng/pngwutil.c")
 
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(arm64|aarch64)")
+    list(APPEND PNG_SRC
+        "libpng/arm/arm_init.c"
+        "libpng/arm/filter_neon_intrinsics.c"
+        "libpng/arm/palette_neon_intrinsics.c")
+  endif()
+
   configure_file("libpng/scripts/pnglibconf.h.prebuilt"
                  "libpng/pnglibconf.h"
                  COPYONLY)

--- a/extern/libpng/pngpriv.h
+++ b/extern/libpng/pngpriv.h
@@ -518,7 +518,7 @@
 #  include <float.h>
 
 #  if (defined(__MWERKS__) && defined(macintosh)) || defined(applec) || \
-    defined(THINK_C) || defined(__SC__) || defined(TARGET_OS_MAC)
+    defined(THINK_C) || defined(__SC__)
    /* We need to check that <math.h> hasn't already been included earlier
     * as it seems it doesn't agree with <fp.h>, yet we should really use
     * <fp.h> if possible.

--- a/extern/zlib/zutil.h
+++ b/extern/zlib/zutil.h
@@ -130,7 +130,7 @@ extern z_const char * const z_errmsg[10]; /* indexed by 2-zlib_error */
 #  endif
 #endif
 
-#if defined(MACOS) || defined(TARGET_OS_MAC)
+#if defined(MACOS)
 #  define OS_CODE  7
 #  ifndef Z_SOLO
 #    if defined(__MWERKS__) && __dest_os != __be_os && __dest_os != __win32_os

